File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/saveprojectaspdf.rb
module Asposetasksjava
  module SaveProjectAsPdf
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        options = Rjb::import('com.aspose.tasks.PdfSaveOptions').new

        # Set the LegendOnEachPage property to false to hide legends

        # Set the row height to fit cell content
        options.setFitContent(true)
        options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').Months)
        options.setPresentationFormat(Rjb::import('com.aspose.tasks.PresentationFormat').TaskUsage)
        options.setLegendOnEachPage(false)
        project.save(data_dir + "project.pdf", options)

        puts "Saved project as pdf, please check the output file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/SaveProjectAsPdf.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\PdfSaveOptions as PdfSaveOptions;
use com\aspose\tasks\Timescale as Timescale;
use com\aspose\tasks\PresentationFormat as PresentationFormat;

class SaveProjectAsPdf{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $options = new PdfSaveOptions();

            # Set the LegendOnEachPage property to false to hide legends

            # Set the row height to fit cell content
        $timescale=new Timescale();
        $presentationFormat=new PresentationFormat();

        $options->setFitContent(true);
        $options->setTimescale($timescale->Months);
        $options->setPresentationFormat($presentationFormat->TaskUsage);
        $options->setLegendOnEachPage(false);
        $project->save($dataDir . "project.pdf", $options);

        print "Saved project as pdf, please check the output file.".PHP_EOL;

    }

}
?>